lib/eventFactory.js

```diff
@@ -505,13 +545,42 @@ async function createStages({ pipeline, groupEventId }) {
         return Promise.resolve();
     }
 
-    const newStages = await convertStages({ pipeline, stages, groupEventId });
+    // list stage names from this pipeline that already exist
+    const existingStages = await stageFactory.list({ params: { pipelineId, name: stageNames } });
+    const existingStageNames = existingStages.map(stage => stage.name);
+    const convertedStages = await convertStages({ pipeline, stages, eventId, jobs, existingStages });
+    const stagesToUpdate = convertedStages.filter(stage => existingStageNames.includes(stage.name));
+    const stagesToCreate = convertedStages.filter(stage => !existingStageNames.includes(stage.name));
     const processed = [];
 
-    // Create new stages
-    newStages.forEach(stage => {
-        logger.info(`Stage:${JSON.stringify(stage)}`);
-        processed.push(stageFactory.create(stage));
+    // Update existing stages
+    stagesToUpdate.forEach(stage => {
+        const updateStageConfig = {};
+
+        if (stage.jobIds) {
+            updateStageConfig.jobIds = stage.jobIds;
+        }
+        if (stage.description) {
+            updateStageConfig.description = stage.description;
+        }
+        if (stage.setup) {
+            updateStageConfig.setup = stage.setup;
+        }
+        if (stage.teardown) {
+            updateStageConfig.teardown = stage.teardown;
+        }
+        if (stage.archived) {
+            updateStageConfig.archived = stage.archived;
+        }
```

Comment:
Can this be simplified?
`archived` should be set to `false` (if the stage was previously removed and now added back in the screwdriver.yaml) 
```suggestion
        const updateStageConfig = {...stage, archived: false};
```

Reply:
can only update certain fields, so can't simplify it that way